cache/remotecache/oss/oss.go

```diff
@@ -0,0 +1,141 @@
+package oss
+
+import (
+	"os"
+	"strings"
+
+	"github.com/aliyun/aliyun-oss-go-sdk/oss"
+	digest "github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+)
+
+const (
+	attrBucket          = "bucket"
+	attrPrefix          = "prefix"
+	attrManifestsPrefix = "manifests_prefix"
+	attrBlobsPrefix     = "blobs_prefix"
+	attrName            = "name"
+	attrEndpointURL     = "endpoint_url"
+	attrAccessKeyID     = "access_key_id"
+	attrSecretAccessKey = "secret_access_key"
+	attrSecurityToken   = "security_token"
+)
+
+type Config struct {
+	Bucket          string
+	Prefix          string
+	ManifestsPrefix string
+	BlobsPrefix     string
+	Names           []string
+	EndpointURL     string
+	AccessKeyID     string
+	SecretAccessKey string
+	SecurityToken   string
+}
+
+func getConfig(attrs map[string]string) (*Config, error) {
+	bucket, ok := attrs[attrBucket]
+	if !ok {
+		bucket, ok = os.LookupEnv("ALIBABA_CLOUD_OSS_BUCKET")
+		if !ok {
+			return &Config{}, errors.Errorf("bucket ($ALIBABA_CLOUD_OSS_BUCKET) not set for oss cache")
+		}
+	}
```

Comment:
Like s3 should we also consider having an env var to set the region? It seems the region is inferred from endpoint url so maybe not needed?: https://www.alibabacloud.com/help/en/oss/developer-reference/go-configure-access-credentials

> // Create an OSSClient instance. 
> // Specify the endpoint of the region in which the bucket is located. For example, if the bucket is located in the China (Hangzhou) region, set the endpoint to https://oss-cn-hangzhou.aliyuncs.com. Specify your actual endpoint.

Just asking because I see the SDK having a method to set it: https://github.com/aliyun/aliyun-oss-go-sdk/blob/2bc58741fa507528ff70b501e7e881a92f88ce3e/oss/client.go#L2891-L2895

Reply:
Yeah, the `region` env var is not needed for OSS SDK